src/Stratis.Bitcoin.Features.Miner/PowBlockDefinition.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+using Stratis.Bitcoin.Mining;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Miner
+{
+    public class PowBlockDefinition : BlockDefinition
+    {
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        // Container for tracking updates to ancestor feerate as we include (parent)
+        // transactions in a block.
+        public class TxMemPoolModifiedEntry
+        {
+            public TxMemPoolModifiedEntry(TxMempoolEntry entry)
+            {
+                this.iter = entry;
+                this.SizeWithAncestors = entry.SizeWithAncestors;
+                this.ModFeesWithAncestors = entry.ModFeesWithAncestors;
+                this.SigOpCostWithAncestors = entry.SigOpCostWithAncestors;
+            }
+
+            public TxMempoolEntry iter;
+
```

Comment:
Is "iter" a standard member name?  will it be used to iterate over something.   the name seems ambiguous.

Reply:
Will revisit in subsequent PRs.